### PR TITLE
Update linker settings to support Clang frameworks

### PR DESCRIPTION
**Pull request by giginet**
I updated some linker settings to avoid some build issues.

- Fix using custom modulemap
- Added `link framework` declare on modulemap for convenience.
- Add dependencies in "Link Binary with Libraries" phase

### CODE REVIEW

File: Sources/ScipioKit/Producer/PIF/ModuleMapGenerator.swift [hunk 1]
```diff
@@ -56,36 +56,45 @@ struct ModuleMapGenerator {
...
+                + generateLinkSection(context: context)
+                + ["}"])
+                .joined()
         }
     }
 
+    private func generateLinkSection(context: Context) -> [String] {
+        context.resolvedTarget.dependencies
+            .compactMap(\.target?.c99name)
+            .map { "    link framework \($0)" }
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/ModuleMapGenerator.swift:95]:
```suggestion
            .map { "    link framework \"\($0)\"" }
```

It needs to add double quote for framework name


```
NIOTLS.xcframework/ios-arm64/NIOTLS.framework/Modules/module.modulemap:4:20: error: expected framework name as a string
    link framework NIO

```

File: Sources/ScipioKit/Producer/PIF/ModuleMapGenerator.swift [hunk 2]
```diff
@@ -56,36 +56,45 @@ struct ModuleMapGenerator {
...
-                    export *
-                }
-                """
-                    .trimmingCharacters(in: .whitespaces)
+                return ([
+                    "framework module \(context.resolvedTarget.c99name) {",
+                    "    umbrella header \"\(headerPath.basename)\"",
+                    "    export *",
+                ]
+                + generateLinkSection(context: context)
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/ModuleMapGenerator.swift:64]: Add `link framework` declares on generated modulemap to be embedded in frameworks.

```
framework module MyModule {
    umbrella header "MyModule.h"
    export *
    link framework "MyDependency1"
    link framework "MyDependency2"
}
```

https://clang.llvm.org/docs/Modules.html#link-declaration

File: Sources/ScipioKit/Producer/PIF/PIFGenerator.swift [hunk 3]
```diff
@@ -260,20 +264,28 @@ private struct PIFLibraryTargetModifier {
         }
         settings[.SWIFT_INSTALL_OBJC_HEADER] = "YES"
 
-        if let clangTarget = resolvedTarget.underlyingTarget as? ClangTarget {
-            switch clangTarget.moduleMapType {
-            case .custom(let moduleMapPath):
-                settings[.MODULEMAP_FILE] = moduleMapPath.moduleEscapedPathString
-                settings[.MODULEMAP_FILE_CONTENTS] = nil
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/PIFGenerator.swift:267]: This overwritten was mistaken.
Respect custom modulemaps to build libraries.

File: Sources/ScipioKit/Logger.swift [hunk 4]
```diff
@@ -38,8 +38,6 @@ struct ScipioLogHandler: LogHandler {
         }
     }
 
-    init() { }
```
Comment (giginet) [inline Sources/ScipioKit/Logger.swift:41]: Just avoid SwiftLint violations

**Review by freddi-kit — APPROVED** [review]

**Pull request merged**